docs/dev-machine-setup.md

```diff
@@ -0,0 +1,109 @@
+# Setting up a development environment for Firecracker
+
+Firecracker currently requires KVM, hence setting up a development environment requires either a physical Linux machine with KVM, or a virtual machine with nested KVM.
+
+## Running Firecracker VMM on GCP with nested KVM
```

Comment:
How about this general structure for this doc:

```
# Setting up a Development Environment for Firecracker

`[Intro Text; Send reader back to getting-started.md once he picks and executes a set of instructions]`

# Local

## Local Bare-Metal Machine

`[TODO]`

## Local Virtual Machine

`[TODO]`

# Cloud

## AWS

`[TODO]`

## GCP

`[Your Instructions]`

### Addendum

`[Your instructions for setting up the account]`

## Microsoft Azure

`[TODO]`
```

Note, I've arranged clouds alphabetically 😄.